src/sage/symbolic/function.pyx

```diff
@@ -230,13 +238,11 @@ cdef class Function(SageObject):
                     callable(getattr(self, real_fname)):
                 raise ValueError(real_fname + " parameter must be callable")
 
-        if not self._is_registered():
-            self._register_function()
-
-            from .expression import symbol_table, register_symbol
-
-            symbol_table['functions'][self._name] = self
+        symbol_table['functions'][self._name] = self
 
```

Comment:
I don't know the mechanics here, but is it right that you don't put ` symbol_table['functions'][self._name] = self` into the body of ` if register_or_update_function`?

Reply:
All instances of each `BuiltinFunction` subclass are equivalent (normally there's only one instance, but more can be created by unpickling.) So unconditionally putting a created instance in the symbol table is correct.
When the symbolic subsystem is not present, there is no way to test `_is_registered`.

Reply:
OK. No problem if you intended.

Reply:
As the keys to the symbol table are not the instances, new items will be added whenever they are created.